apps/api/src/services/data-provider/data-enhancer/trackinsight/trackinsight.service.ts

```diff
@@ -180,4 +163,51 @@ export class TrackinsightDataEnhancerService implements DataEnhancerInterface {
   public getTestSymbol() {
     return 'QQQ';
   }
+
+  private async searchTrackinsightSymbol({
+    requestTimeout = this.configurationService.get('REQUEST_TIMEOUT'),
+    symbol
+  }: {
+    symbol: string;
+    requestTimeout: number;
+  }) {
+    const newSymbol = await fetch(
+      `https://www.trackinsight.com/search-api/search_v2/${symbol}/_/ticker/default/0/3`,
+      {
+        signal: AbortSignal.timeout(requestTimeout)
+      }
+    )
+      .then((res) => res.json())
+      .then((jsonRes) => {
+        if (jsonRes['results']['count'] === 1) {
+          // Return the only ticker that matches the one in the search
+          return jsonRes['results']['docs'][0]['ticker'];
+        }
+        return undefined;
+      })
+      .catch(() => {
+        return undefined;
+      });
+    if (newSymbol) {
+      return newSymbol;
+    }
+
+    return await fetch(
```

Comment:
This duplicates the request above, consider refactoring to avoid repetition (call `searchTrackinsightSymbol()` twice in `enhance()` with different `symbol` parameter)

Reply:
> Would something like this be okay?

At first glance, that's fine 👍🏻 I will have a closer look later.